apps/settings/js/modules/dialog_panel.js

```diff
@@ -3,17 +3,48 @@ define(function(require) {
 
   var Panel = require('modules/panel');
 
-  var DialogPanel = function ctor_dialogPanel(options) {
+  var DialogPanel = function ctor_dialogPanel(panelOptions) {
+    var panel = Panel(panelOptions);
+
     var _emptyFunc = function() {};
-    var panel = Panel(options);
+    var submitFunction = _emptyFunc;
+    var cancelFunction = _emptyFunc;
+
+    // Because we need to bridge onWrapSubmit & onWrapCancel with panelOptions
+    // together, we have to override panel.init() to make sure we can access
+    // these API here.
+    //
+    // When init() is called, we will automatically make submitFunction &
+    // cancelFunction's reference work and link to onWrapSubmit & onWrapCancel,
+    // in this way, we can call `this.submit()` & `this.cancel()` in
+    // panelOptions.
+    var originalInit = panel.init;
+    panel.init = function(panel, userOptions) {
```

Comment:
There are two different `panel`s here. I would suggest to rename the second one to `panelElement` to avoid possible confusion.